addons/agent_mirrorpeer_controller.go

```diff
@@ -147,6 +148,34 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
+	// Set keyRotation type on SC based on the value of mirrorpeer annotation for "async" DR type only
+	if mirrorPeer.Spec.Type == multiclusterv1alpha1.Async {
+		spokeKeyTypeAnn := fmt.Sprintf("%s.%s", r.SpokeClusterName, SpokeKeyTypeAnnotation)
+		sc, err := utils.FetchStorageCluster(ctx, r.SpokeClient, scr.Name, scr.Namespace)
+		if err != nil {
+			logger.Error("Failed to fetch storagecluster", "error", err)
+			return ctrl.Result{}, err
+		}
+
+		spokeAnnVal := utils.KeyTypeAES
+		if mirrorPeer.Annotations[utils.KeyTypeAnnotation] != utils.KeyTypeAES256k {
+			if sc.Status.Version >= version.Version {
+				spokeAnnVal = utils.KeyTypeAES256k
+			}
+			if utils.AddAnnotation(&mirrorPeer, spokeKeyTypeAnn, spokeAnnVal) {
+				if err = r.HubClient.Update(ctx, &mirrorPeer); err != nil {
+					logger.Error("Failed to update spoke cluster keyType annotation on MirrorPeer", "error", err)
+					return ctrl.Result{}, err
+				}
+			}
+		}
+
+		if err = utils.SetKeyTypeOnStorageCluster(ctx, logger, r.SpokeClient, r.HubClient, sc); err != nil {
```

Comment:
We are calling SetKeyTypeOnStorageCluster twice, once in agent_mirrorpeer_controller and once in storagecluster controller

Reply:
Yes, because the `storagecluster_controller` will only be invoked if there are any annotation changes like if someone removes the annotation. We have a predicate to check if the old != new, only then proceed with reconcile.
And, in the `agent_mirrorpeer_controller` we need it after initial reconcile to set the annotation at the first place.

Reply:
Mentioned the reason in above [comment](https://github.com/red-hat-storage/odf-multicluster-orchestrator/pull/340#discussion_r2332594289)